pb-jelly-gen does not work on Native Windows

Creating a -gen create using pb-jelly-gen does not work on Windows due to unix-specific imports.

Output of `cargo run --bin proto-gen` (proto-gen is the gen crate in a cargo workspace):

```
   Compiling pb-jelly-gen v0.0.4
error[E0433]: failed to resolve: could not find `unix` in `os`
  --> <home dir>\.cargo\registry\src\github.com-1ecc6299db9ec823\pb-jelly-gen-0.0.4\src\lib.rs:42:9
   |
42 |     os::unix::fs::PermissionsExt,
   |         ^^^^ could not find `unix` in `os`

error[E0599]: no method named `set_mode` found for struct `std::fs::Permissions` in the current scope
   --> <home dir>\.cargo\registry\src\github.com-1ecc6299db9ec823\pb-jelly-gen-0.0.4\src\lib.rs:273:29
    |
273 |                 permissions.set_mode(0o777);
    |                             ^^^^^^^^ method not found in `std::fs::Permissions`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `pb-jelly-gen`.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,8 +6,7 @@ env: RUSTC_WRAPPER=sccache
 os:
   - linux
   - osx
-  # Coming soon to a CI near you
-  # - windows
+  - windows
 addons:
   apt:
     packages:
@@ -21,6 +20,16 @@ before_install:
     - |
       if [ "$TRAVIS_OS_NAME" = "windows" ]; then
         PROTOC_FILENAME=protoc-${PROTOBUF_VERSION}-win64.zip
+        choco install python --version=3.8.6
+        choco install protoc --version=3.12.3
+        choco install sccache --version=0.2.12
+        export PATH="/c/Python38:/c/Python38/Scripts:$PATH"
+        echo $PATH
+        cp /c/Python38/python.exe /c/Python38/python3.exe
+        ls /c/Python38
+        ls /c/Python38/Scripts
+        python3.exe --version
+        python3 --version
       else
         if [ "$TRAVIS_OS_NAME" = "osx" ]; then
           # Brew seems to have an older version installed - and for some reason, python3
@@ -48,6 +57,7 @@ before_install:
     - python3 -m pip install -r pb-jelly-gen/requirements.txt
     - python3 -c "import google.protobuf ; print(google.protobuf.__version__)"
     - python3 -m pip install mypy==0.782 mypy-protobuf==1.23
+    - rustup default nightly-2020-10-24
     - rustup component add rustfmt
 script:
     - protoc --version

--- a/pb-jelly-gen/src/lib.rs
+++ b/pb-jelly-gen/src/lib.rs
@@ -34,12 +34,13 @@ use include_dir::{
     include_dir,
     Dir,
 };
+#[cfg(not(windows))]
+use std::os::unix::fs::PermissionsExt;
 use std::{
     convert::AsRef,
     fs,
     io::Write,
     iter::IntoIterator,
-    os::unix::fs::PermissionsExt,
     path::{
         Path,
         PathBuf,
@@ -197,7 +198,9 @@ impl GenProtos {
 
     fn gen_rust_protos(&self, temp_dir: tempfile::TempDir) -> Output {
         // Temp path to the codegen script
-        let codegen_path = temp_dir.path().join("codegen.py");
+        let codegen_path = temp_dir
+            .path()
+            .join(if cfg!(windows) { "codegen.bat" } else { "codegen.py" });
 
         let mut protoc_cmd = Command::new("protoc");
 
@@ -269,12 +272,15 @@ impl GenProtos {
                 let mut abs_file = fs::OpenOptions::new().write(true).create_new(true).open(&abs_path)?;
                 abs_file.write_all(file.contents())?;
 
-                let mut permissions = abs_file.metadata()?.permissions();
-                permissions.set_mode(0o777);
-                drop(abs_file);
+                #[cfg(not(windows))]
+                {
+                    let mut permissions = abs_file.metadata()?.permissions();
+                    permissions.set_mode(0o777);
+                    drop(abs_file);
 
-                // Set permissions of the file so it is executable
-                fs::set_permissions(&abs_path, permissions)?;
+                    // Set permissions of the file so it is executable
+                    fs::set_permissions(&abs_path, permissions)?;
+                }
             }
 
             for dir in dir.dirs() {

--- a/pb-test/src/verify_generated_files.rs
+++ b/pb-test/src/verify_generated_files.rs
@@ -1,5 +1,7 @@
 use std::env;
 use std::fs;
+
+use pretty_assertions::assert_eq;
 use walkdir::WalkDir;
 
 /// Ensure that generated files match .expected
@@ -35,8 +37,7 @@ fn verify_generated_files() {
 
         let contents = fs::read_to_string(proto_file).unwrap();
         let expected_contents = fs::read_to_string(&expected).unwrap();
-        if contents != expected_contents {
-            panic!(".expected files don't match - Please run `cd pb-test ; cargo test` to generate");
-        }
+        assert_eq!(contents.lines().collect::<Vec<_>>(), expected_contents.lines().collect::<Vec<_>>(),
+                   ".expected files don't match - Please run `cd pb-test ; cargo test` to generate");
     }
 }

